### DISCUSSION TITLE
add custom option

### DISCUSSION THREAD

**Discussion by hardin90** [seleniumbase/SeleniumBase#922] [Q&A] [answered]
How i can add custom options to start tests, like `pytest test_1.py --my-custom-option`

**Comment by mdmintz** [comment, marked as answer]
@hardin90 
There are some custom built-in command-line args that you can use:
![D511E966-4615-43E3-8685-30D6BCEFA580](https://user-images.githubusercontent.com/6788579/123551626-a3263100-d740-11eb-8ed0-44432bd1921c.jpeg)

**Reply by MovingW** [reply]
How could I add `"goog:loggingPrefs", {"performance": "ALL", "browser": "ALL"}` in this case?

**Reply by mdmintz** [reply]
That option is already set if `uc_cdp` is True.
https://github.com/seleniumbase/SeleniumBase/blob/39cf213bded5f5873c93f76e31586ac4f40948e2/seleniumbase/undetected/__init__.py#L166

**Reply by MovingW** [reply]
But the uc_cdp mode also bring new issue. I could not fetch all logs at one time and it ONLY listen to the active window, not the whole browser, which means if the page is opened in a new tab, the logs would be gone.

**Reply by mdmintz** [reply]
This works:

```python
from seleniumbase import Driver

driver = Driver(uc=True, uc_cdp=True)
try:
    driver.get("https://seleniumbase.io")
    print(driver.get_log("performance"))
finally:
    driver.quit()
```

Prints the logs. It should only work on the currently active window. Otherwise UC Mode would be detected.

**Reply by MovingW** [reply]
But if you add a time.sleep(3) before `driver.get_log("performance")`, you'll get nothing. For all logs have been fetched by the background cdp listeners(even if you never add one in your code).

**Reply by mdmintz** [reply]
Resolved in `4.20.8` - https://github.com/seleniumbase/SeleniumBase/issues/2220
